Ao/payment

AleOshiro opened the pull request:
### 2023.08.17
- feat: Se agrega nuevo controlador Payment con su respectiva documentación y servicio (6bf1281)
- feat: Se agregan las clases DTO para la entidad Payment (aeba1e9)
- feat: Se agrega servicio gestiona los pagos (969474a)
- refactor: Se agregan nuevas variables de configuración que definen el tiempo limite de pago según la forma de pago elegida por el usuario. (790f5a8)
- feat: agrego clase Tools que contiene método que retorna la fecha y hora actual (e81ab4d)
- refactor: Se agrega la interfaz de PaymentRepository (5df6b32)
- refactor: Se agrega la relación de la entidad Booking con la entidad Payment (fff1292)
- feat: Se agrega nueva entidad Payment y enums donde se define "la forma de pago" y "el estado del pago" (646bb82)

AleOshiro committed:
feat: agrego nueva entidad Payment y enums de estados

feat: Se agrega nueva entidad Payment y enums donde se define "la forma de pago" y "el estado del pago"

AleOshiro committed:
refactor: agrego la relación de Payment con Booking

refactor: Se agrega la relación de la entidad Booking con la entidad Payment

AleOshiro committed:
refactor: agrego la interfaz de PaymentRepository

refactor: Se agrega la interfaz de PaymentRepository

AleOshiro committed:
feat: agrego clase utils.Tools

feat: agrego clase Tools que contiene método que retorna la fecha y hora actual

AleOshiro committed:
refactor: agrego nuevas variables de configuración

refactor: Se agregan nuevas variables de configuración que definen el tiempo limite de pago según la forma de pago elegida por el usuario.

AleOshiro committed:
feat: agrego servicio Payment

feat: Se agrega servicio gestiona los pagos

AleOshiro committed:
docs: actualizo archivo CHANGELOG.md

keynaka commented on src/main/resources/application-h2.yml:
```diff
@@ -18,4 +18,7 @@ spring:
-    defaultModelsExpandDepth: -1 # Hide the model section
\ No newline at end of file
+    defaultModelsExpandDepth: -1 # Hide the model section
+paymentTimeLimitFor:
```
esto veo que es algo que no depende del profile que definamos no? SI es algo que es siempre igual, creo que conviene que este dentro de application.yml que es el comun para todos. Siempre busca en el application-{profile},yml y si no lo encuentra busca en el general q es el application.yml

AleOshiro replied:
Los test que esta haciendo necesitan de h2, no? Porque justo en el branch anterior te borre el application.yml :P

- Olvidate lo dije jaja

AleOshiro committed:
refactor: agrego application.yml con la config. general del proyecto

refactor: Se agrega application.yml con la configuración general que utiliza el proyecto

AleOshiro committed:
feat: agrego clases Dto para Payment

feat: Se agregan las clases DTO para la entidad Payment

AleOshiro committed:
feat: agrego nuevo controlador Payment

feat: Se agrega nuevo controlador Payment con su respectiva documentación y servicio

AleOshiro committed:
docs: actualizo archivo CHANGELOG.md

AleOshiro committed:
build: agrego dependencia para PostgreSQL

build: Se agrega dependencia para PostgreSQL

AleOshiro committed:
feat: agrego implementación para el cambio de estado de un pago y de una reserva

feat: Se agrega implementación para el cambio de estado de un pago y de una reserva

AleOshiro committed:
docs: actualizo archivo CHANGELOG.md

AleOshiro committed:
feat: agrego implementación del cambio de estado de un pago, reserva y butaca

feat: Se agrega implementación del cambio de estado de un pago, reserva y butaca

AleOshiro committed:
refactor: quito método de '.getBalance()' del controller Payment

refactor: Se quita método de '.getBalance()' del controller Payment para migrarlo a otro controller

AleOshiro committed:
docs: actualizo archivo CHANGELOG.md
